examples/gtc_final_report_matrix.py

```diff
@@ -0,0 +1,146 @@
+from numpy import around
+
+"""
+This example will allow the user to create matrix report files identical to the ones exported from GenomeStudio
+by chosing the genotype with or without the genotype scores.
+"""
+
+def build_dict(matrix = None, samplename = None, names = None, genotypes = None, genotype_scores = None):
+    if genotype_scores is not None:
+        for (name, genotype, genotype_score) in zip(names, genotypes, genotype_scores):
+            if samplename in matrix:
+                matrix[samplename][name] = genotype + "|" + str(genotype_score)
+            else:
+                matrix[samplename] = {}
+                matrix[samplename][name] = genotype + "|" + str(genotype_score)
+    else:
+        for (name, genotype) in zip(names, genotypes):
+            if samplename in matrix:
+                matrix[samplename][name] = genotype
+            else:
+                matrix[samplename] = {}
+                matrix[samplename][name] = genotype
+    return matrix
+
+delim = "\t"
+NUM_ARGUMENTS = 6
+
+parser = argparse.ArgumentParser("Generate a final report from a directory of GTC files")
+parser.add_argument("manifest", help="BPM manifest file")
+parser.add_argument("gtc_directory", help="Directory containing GTC files")
+parser.add_argument("output_file", help="Location to write report")
+parser.add_argument("--forward", help="python gtc_final_report_matrix.py <path_to_manifest> <path_to_gtc_directory> <path_to_output_file> --forward 1, print matrix with forward alleles")
+parser.add_argument("--forward_GC", help="python gtc_final_report_matrix.py <path_to_manifest> <path_to_gtc_directory> <path_to_output_file> --forward_GC 1, print matrix with forward alleles including genotype scores")
+parser.add_argument("--top", help="python gtc_final_report_matrix.py <path_to_manifest> <path_to_gtc_directory> <path_to_output_file> --top 1, print matrix with top alleles")
+parser.add_argument("--top_GC", help="python gtc_final_report_matrix.py <path_to_manifest> <path_to_gtc_directory> <path_to_output_file> --top_GC 1, print matrix with top alleles including genotype scores")
+parser.add_argument("--AB", help="python gtc_final_report_matrix.py <path_to_manifest> <path_to_gtc_directory> <path_to_output_file> --forward 1, print matrix with forward alleles")
+parser.add_argument("--AB_GC", help="python gtc_final_report_matrix.py <path_to_manifest> <path_to_gtc_directory> <path_to_output_file> --forward_GC 1, print matrix with forward alleles including genotype scores")
+parser.add_argument("--plus", help="python gtc_final_report_matrix.py <path_to_manifest> <path_to_gtc_directory> <path_to_output_file> --top 1, print matrix with top alleles")
+parser.add_argument("--plus_GC", help="python gtc_final_report_matrix.py <path_to_manifest> <path_to_gtc_directory> <path_to_output_file> --top_GC 1, print matrix with top alleles including genotype scores")
+
+args = parser.parse_args()
+
+if len(sys.argv) != NUM_ARGUMENTS:
```

Comment:
The argument parser should be able to handle this error?

Reply:
In this case yes as only one matrix report is supported per run, also didn't make sense to me to add a default case.